naive_ralloc/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+#![feature(allocator)]
+#![feature(linkage)]
+#![no_std]
+#![allocator]
+
+#[macro_use]
+extern crate sc;
+
+use core::ptr;
+
+// fs/mmap.c
+pub unsafe fn brk(brk: usize) -> usize {
+    syscall!(BRK, brk)
+}
+
+fn allocate(size: usize) -> *mut u8 {
```

Comment:
`sbrk` isn't a syscall but some libc function, from the same man page:

>On Linux, sbrk() is implemented as a library function that uses the brk() system call, and does some internal bookkeeping so that it can return the old break value.

`brk` gets called with a higher memory address than before, therefore allocating memory.